### PR TITLE
Fix external ports

### PR DESCRIPTION
**Pull request by JakeSCahill**
Also added a sample for a single-broker cluster.

### CODE REVIEW

File: docs/get-started/quick-start/quick-start-docker.mdx [hunk 1]
```diff
@@ -39,6 +39,131 @@ You should see the version of Docker Compose that's installed on your local mach
...
+        # address the broker advertises to clients that connect to the Kafka API.
+        - --advertise-kafka-addr
+        - internal://redpanda-0:9092,external://localhost:19092
+        - --pandaproxy-addr
+        - internal://0.0.0.0:8082,external://0.0.0.0:18082
+        # address the broker advertises to clients that connect to PandaProxy.
+        - --advertise-pandaproxy-addr
+        - internal://redpanda-0:8082,external://localhost:18082
+        - --schema-registry-addr
+        - redpanda-0:8081
```
Comment (BenPope) [inline docs/get-started/quick-start/quick-start-docker.mdx:88]: Let's add an internal and external listener for schema registry (no advertised required):
```suggestion
        - internal://redpanda-0:8081,external://0.0.0.0:18081
```

File: docs/get-started/quick-start/quick-start-docker.mdx [hunk 2]
```diff
@@ -39,6 +39,131 @@ You should see the version of Docker Compose that's installed on your local mach
...
+      container_name: redpanda-0
+      volumes:
+        - redpanda-0:/var/lib/redpanda/data
+      networks:
+        - redpanda_network
+      ports:
+        - 18081:18081
+        - 18082:18082
+        - 19092:19092
+        - 19644:19644
```
Comment (BenPope) [inline docs/get-started/quick-start/quick-start-docker.mdx:116]: I don't recall if we can add multiple listeners for the admin API, and I don't think rpk redpanda start supports it directly, but perhaps this is better:
```suggestion
        - 19644:9644
```

File: docs/get-started/quick-start/quick-start-docker.mdx [hunk 3]
```diff
@@ -39,6 +39,131 @@ You should see the version of Docker Compose that's installed on your local mach
...
+        - 8081:8081
+    console:
+      container_name: redpanda-console
+      image: docker.redpanda.com/vectorized/console:v2.1.1
```
Comment (BenPope) [inline docs/get-started/quick-start/quick-start-docker.mdx:120]: Isn't this version a little broken? I thought we had a newer version incoming (you'll want to check this):
```suggestion
      image: docker.redpanda.com/vectorized/console:v2.1.0
```

**Review by BenPope — COMMENTED** [review]
Minor suggestions, please do the same for the three node.

File: docs/get-started/quick-start/quick-start-docker.mdx [hunk 4]
```diff
@@ -39,6 +39,131 @@ You should see the version of Docker Compose that's installed on your local mach
...
+      volumes:
+        - redpanda-0:/var/lib/redpanda/data
+      networks:
+        - redpanda_network
+      ports:
+        - 18081:18081
+        - 18082:18082
+        - 19092:19092
+        - 19644:9644
+        - 8081:8081
```
Comment (BenPope) [inline docs/get-started/quick-start/quick-start-docker.mdx:117]: 8081 is not needed, 18081 should be used instead.

File: docs/get-started/quick-start/quick-start-docker.mdx [hunk 5]
```diff
@@ -176,10 +303,11 @@ You should see the version of Docker Compose that's installed on your local mach
...
         - redpanda_network
       ports:
-        - 38081:8081
-        - 38082:8082
-        - 39092:9092
+        - 38081:38081
+        - 38082:38082
+        - 39092:39092
         - 39644:9644
+        - 8083:8081
```
Comment (BenPope) [inline docs/get-started/quick-start/quick-start-docker.mdx:310]: 8083 is not needed, 38081 should be used instead.

File: docs/get-started/quick-start/quick-start-docker.mdx [hunk 6]
```diff
@@ -139,10 +265,11 @@ You should see the version of Docker Compose that's installed on your local mach
...
         - redpanda_network
       ports:
-        - 28081:8081
-        - 28082:8082
-        - 29092:9092
+        - 28081:28081
+        - 28082:28082
+        - 29092:29092
         - 29644:9644
+        - 8082:8081
```
Comment (BenPope) [inline docs/get-started/quick-start/quick-start-docker.mdx:272]: 8082 is not needed, 28081 should be used instead.

File: docs/get-started/quick-start/quick-start-docker.mdx [hunk 7]
```diff
@@ -104,10 +229,11 @@ You should see the version of Docker Compose that's installed on your local mach
...
         - redpanda_network
       ports:
-        - 18081:8081
-        - 18082:8082
-        - 19092:9092
+        - 18081:18081
+        - 18082:18082
+        - 19092:19092
         - 19644:9644
+        - 8081:8081
```
Comment (BenPope) [inline docs/get-started/quick-start/quick-start-docker.mdx:236]: 8081 is not needed, 18081 should be used instead.

File: docs/get-started/quick-start/quick-start-docker.mdx [hunk 8]
```diff
@@ -403,10 +534,23 @@ docker compose down
...
+
+</TabItem>
+<TabItem value="three" label="Three Brokers">
+
 ```bash
 docker volume rm redpanda-quickstart_redpanda-0 redpanda-quickstart_redpanda-1 redpanda-quickstart_redpanda-2
 ```
 
+</TabItem>
+</Tabs>
```
Comment (BenPope) [inline docs/get-started/quick-start/quick-start-docker.mdx:552]: Lets simplify this:
~~~suggestion
```bash
docker compose down -v
```
~~~

**Review by BenPope — COMMENTED** [review]
Looks pretty good.

File: docs/get-started/quick-start/quick-start-docker.mdx (hunk 7 above)
Comment (JakeSCahill) [inline docs/get-started/quick-start/quick-start-docker.mdx:236]:
```suggestion
```

**Review by Feediver1 — APPROVED** [review]
lgtm

File: docs/get-started/quick-start/quick-start-docker.mdx (hunk 6 above)
Comment (JakeSCahill) [inline docs/get-started/quick-start/quick-start-docker.mdx:272]:
```suggestion
```

File: docs/get-started/quick-start/quick-start-docker.mdx (hunk 5 above)
Comment (JakeSCahill) [inline docs/get-started/quick-start/quick-start-docker.mdx:310]:
```suggestion
```

File: versioned_docs/version-22.3/get-started/quick-start/quick-start-docker.mdx [hunk 9]
```diff
@@ -39,6 +39,131 @@ You should see the version of Docker Compose that's installed on your local mach
...
+      volumes:
+        - redpanda-0:/var/lib/redpanda/data
+      networks:
+        - redpanda_network
+      ports:
+        - 18081:18081
+        - 18082:18082
+        - 19092:19092
+        - 19644:9644
+        - 8081:8081
```
Comment (JakeSCahill) [inline versioned_docs/version-22.3/get-started/quick-start/quick-start-docker.mdx:117]:
```suggestion
```

File: versioned_docs/version-22.3/get-started/quick-start/quick-start-docker.mdx [hunk 10]
```diff
@@ -104,10 +229,11 @@ You should see the version of Docker Compose that's installed on your local mach
...
         - redpanda_network
       ports:
-        - 18081:8081
-        - 18082:8082
-        - 19092:9092
+        - 18081:18081
+        - 18082:18082
+        - 19092:19092
         - 19644:9644
+        - 8081:8081
```
Comment (JakeSCahill) [inline versioned_docs/version-22.3/get-started/quick-start/quick-start-docker.mdx:236]:
```suggestion
```

File: versioned_docs/version-22.3/get-started/quick-start/quick-start-docker.mdx [hunk 11]
```diff
@@ -139,10 +265,11 @@ You should see the version of Docker Compose that's installed on your local mach
...
         - redpanda_network
       ports:
-        - 28081:8081
-        - 28082:8082
-        - 29092:9092
+        - 28081:28081
+        - 28082:28082
+        - 29092:29092
         - 29644:9644
+        - 8082:8081
```
Comment (JakeSCahill) [inline versioned_docs/version-22.3/get-started/quick-start/quick-start-docker.mdx:272]:
```suggestion
```

File: versioned_docs/version-22.3/get-started/quick-start/quick-start-docker.mdx [hunk 12]
```diff
@@ -176,10 +303,11 @@ You should see the version of Docker Compose that's installed on your local mach
...
         - redpanda_network
       ports:
-        - 38081:8081
-        - 38082:8082
-        - 39092:9092
+        - 38081:38081
+        - 38082:38082
+        - 39092:39092
         - 39644:9644
+        - 8083:8081
```
Comment (JakeSCahill) [inline versioned_docs/version-22.3/get-started/quick-start/quick-start-docker.mdx:310]:
```suggestion
```

File: versioned_docs/version-22.3/get-started/quick-start/quick-start-docker.mdx [hunk 13]
```diff
@@ -403,10 +534,23 @@ docker compose down
...
+
+</TabItem>
+<TabItem value="three" label="Three Brokers">
+
 ```bash
 docker volume rm redpanda-quickstart_redpanda-0 redpanda-quickstart_redpanda-1 redpanda-quickstart_redpanda-2
 ```
 
+</TabItem>
+</Tabs>
```
Comment (JakeSCahill) [inline versioned_docs/version-22.3/get-started/quick-start/quick-start-docker.mdx:552]:
```suggestion
```bash
docker compose down -v
```
```

**Review by BenPope — APPROVED** [review]
LGTM

I never really like the example of running rpk in one of the redpanda containers, I think it would be better as  it's own container in the docker network, but that's a different change.

**Comment by JakeSCahill** [comment]
> LGTM
> 
> I never really like the example of running rpk in one of the redpanda containers, I think it would be better as it's own container in the docker network, but that's a different change.

We have an example of this here: https://docs.redpanda.com/docs/reference/docker-compose/#redpanda-enterprise-edition

For this quickstart, I think it's useful to show how to do ad-hoc rpk commands on the containers. But maybe we can show both approaches in a future change.

**Pull request merged**